client/deployments/client.go

```diff
@@ -49,15 +88,92 @@ func NewClient(url string, skipVerify bool) *Client {
 
 	return &Client{
 		url:               url,
-		artifactUploadUrl: JoinURL(url, artifactUploadUrl),
+		artifactUploadURL: JoinURL(url, artifactUploadURL),
+		artifactsListURL:  JoinURL(url, artifactsListURL),
 		client: &http.Client{
 			Transport: tr,
 		},
 	}
 }
 
-func (c *Client) UploadArtifact(description, artifactPath, tokenPath string, noProgress bool) error {
+func (c *Client) ListArtifacts(tokenPath string, detailLevel int) error {
+	token, err := ioutil.ReadFile(tokenPath)
+	if err != nil {
+		return errors.Wrap(err, "Please Login first")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, c.artifactsListURL, nil)
+	if err != nil {
+		return errors.Wrap(err, "Cannot create request")
+	}
+	req.Header.Set("Authorization", "Bearer "+string(token))
+
+	reqDump, _ := httputil.DumpRequest(req, false)
+	log.Verbf("sending request: \n%v", string(reqDump))
+
+	rsp, err := c.client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "Get /deployments/artifacts request failed")
```

Comment:
Suggestion:
Also handle the HTTP error codes returned. As this is not an error. The request in itself is succesful, therefore the HTTP status code needs to be checked here, otherwise there will be no data to unmarshal in the case of HTTP error codes with no json in the body response.

Reply:
@siredmar what are your thoughts on this one?

Reply:
Oh i missed that one. Sure, check != 200?

Reply:
https://docs.mender.io/api/#list-releases-responses

Yup, looks like that should do the trick for now :)